java/src/org/certificatetransparency/ctlog/serialization/CryptoDataLoader.java

```diff
@@ -83,21 +83,28 @@ static PublicKey parsePublicKey(List<String> pemLines) {
     String b64string = Joiner.on("").join(pemLines.subList(1, pemLines.size() - 1));
     // Extract public key
     X509EncodedKeySpec spec = new X509EncodedKeySpec(Base64.decodeBase64(b64string));
-    // Note: EC KeyFactory does not exist in openjdk, only Oracle's JDK.
-    KeyFactory kf = null;
+    KeyFactory kf;
     try {
-      kf = KeyFactory.getInstance("EC");
-      return kf.generatePublic(spec);
-    } catch (NoSuchAlgorithmException e) {
-      // EC is known to be missing from openjdk; Oracle's JDK must be used.
-      throw new UnsupportedCryptoPrimitiveException("EC support missing", e);
-    } catch (InvalidKeySpecException e) {
-      throw new InvalidInputException("Log public key is invalid", e);
+        // First try to parse as an RSA key. If it's an EC key then generatePublic throws.
+        // On OpenJDK it throws NPE, but a different exception might be thrown on other Java implementations
+        kf = KeyFactory.getInstance("RSA");
```

Comment:
Will have a look at it tomorrow. I looked quickly at the OpenJDK source now, and it looks like it always returns the string "X.509". So not sure if it will work, but if not I'll try to find some alternative